### ISSUE TITLE
failed lcms2 download for OSX 10.11

### ISSUE THREAD

**Issue by newser1** [Homebrew/homebrew-science#3091]
I get the error msg: Download failed: https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz. Looking up Sourceforge, it lists a slightly different path to the file: http://sourceforge.net/projects/lcms/files/lcms/2.7/lcms2-2.7.tar.gz/download, with the "files" subdirectory missing from the gitHub path, is gitHub just using a different file source with the same file or is this reason for failed download termination, if yes how can I correct this?

bash-3.2$ HOMEBREW_MAKE_JOBS=1 brew install -v octave 2>&1
==> Installing octave from homebrew/homebrew-science
==> Installing dependencies for homebrew/science/octave: little-cms2, ghostscript, epstool, libpng, gl2ps, libtool, freetype, graphicsmagick, fftw, glpk, fontconfig, gd, lua, gnuplot, szip, hdf5, plotutils, imagemagick, pstoedit, cmake, qhull, qrupdate, tbb, suite-sparse421
==> Installing homebrew/science/octave dependency: little-cms2
==> Downloading https://homebrew.bintray.com/bottles/little-cms2-2.7.el_capitan.bottle.tar.gz
/usr/bin/curl -fLA Homebrew 0.9.5 (Ruby 2.0.0-645; OS X 10.11.1) https://homebrew.bintray.com/bottles/little-cms2-2.7.el_capitan.bottle.tar.gz -C 0 -o /Library/Caches/Homebrew/little-cms2-2.7.el_capitan.bottle.tar.gz.incomplete
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (60) SSL certificate problem: Invalid certificate chain
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
Error: Failed to download resource "little-cms2"
Download failed: https://homebrew.bintray.com/bottles/little-cms2-2.7.el_capitan.bottle.tar.gz
Warning: Bottle installation failed: building from source.
==> Downloading https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz
/usr/bin/curl -fLA Homebrew 0.9.5 (Ruby 2.0.0-645; OS X 10.11.1) https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz -C 0 -o /Library/Caches/Homebrew/little-cms2-2.7.tar.gz.incomplete
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (60) SSL certificate problem: Invalid certificate chain
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
Error: Failed to download resource "little-cms2"
Download failed: https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz

https://gist.github.com/newser1/bb36a023809dee3f8ec6

**Comment by dpo** [comment]
Please don't open multiple issues about the same problem. 

Are you located behind a firewall?

**Comment by newser1** [comment]
Sorry. I'm a first time user, made several attempts and sent the same issue twice.

Received msg(rm: /Library/Caches/Homebrew/little-cms2-2.*: No such file or directory) twice, both before and after running (brew install lcms2).

Only firewall running is one supplied by my OSX10.11  Why do the other dependency downloads up to this pt in the install do fine?  What should I do next?

**Comment by newser1** [comment]
running brew doctor:

bash-3.2$ brew doctor
Your system is ready to brew.

**Comment by dpo** [comment]
Could you try `brew upgrade curl`?

**Comment by newser1** [comment]
Getbash-3.2$ brew upgrade curl
Error: curl not installed
I guess this is the cause, should curl be part of the brew install tools or is it part of Xcode or Mac command tools.  Looking back at the original error msg, it does have a long discussion of a curl SSL certificate problem (never encountered certificate authorities in context of compilations before).  Do I need to install curl via brew or what?
 


    On Saturday, December 5, 2015 12:37 AM, Dominique <notifications@github.com> wrote:
 

 Could you try brew upgrade curl?—

**Comment by newser1** [comment]
(brew upgrade curl) gets (curl not installed)

so I (brew install curl), now I get (curl 7.46.0 already installed). in response to (brew upgrade curl).

after the compile and install, the msg  s that ( This formula is keg-only, and not lsymlinked into /usr/ocal.  OSX already provides curl but this in parallel can cause "all kinds of trouble",)  But it claims there are no consequences for me, but to use it to build software I have to add to build variables:
              LDFLAGS:  -L/usr/local/opt/curl/lib
              CPPFLAGS:  -I/usr/local/opt/curl/include

     zsh completion has been installed to:
     /usr/local/share/zsh/site-functions

what and how should I proceed now, should I remove curl or modify the 2 FLAG variables.  I don't want to create "all kinds of trouble" for using Xcode in the future.

Thanks
--------------------------------------------
On Fri, 12/4/15, Dominique <notifications@github.com> wrote:

 Subject: Re: [homebrew-science] failed lcms2 download for OSX 10.11 (#3091)
 To: "Homebrew/homebrew-science" <homebrew-science@noreply.github.com>
 Cc: "newser1" <pywus@yahoo.com>
 Date: Friday, December 4, 2015, 9:37 PM
 
 Could you try `brew upgrade curl`?
 
 ---

**Comment by dpo** [comment]
Please try `brew install lcms2` now.

**Comment by newser1** [comment]
now I have curl but ran >brew install lcms2
but still failed with same msg of:    Error: Failed to download resource "little-cms2"
    Download failed: https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz
Terminal output:
 bash-3.2$ brew install lcms2
 ==> Downloading https://homebrew.bintray.com/bottles/little-cms2-2.7.el_capitan.bottle.tar.gz

curl: (60) SSL certificate problem: Invalid certificate chain
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
Error: Failed to download resource "little-cms2"
Download failed: https://homebrew.bintray.com/bottles/little-cms2-2.7.el_capitan.bottle.tar.gz
Warning: Bottle installation failed: building from source.
==> Downloading https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz

curl: (60) SSL certificate problem: Invalid certificate chain
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
Error: Failed to download resource "little-cms2"
Download failed: https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz

--mm strange, I downloaded the latest Xcode and Terminal command tools just before attempt installing Octave, I assume brew's unable to use the curl supplied with Xcode

maybe I still need to set the 2 flags (LDFLAGS & CPPFLAGS) mentioned in my last msg to get curl working for the download to succeed--how do i set the flags, what do I have to type in Term

 


    On Saturday, December 5, 2015 12:21 PM, Dominique <notifications@github.com> wrote:
 

 Please try brew install lcms2 now.—

**Comment by dpo** [comment]
Does it work from the command line (i.e., no Homebrew involved)?
```
curl https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz -O
```
or
```
/usr/local/opt/curl/bin/curl https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz -O
```
If not, please check your firewall settings.

**Comment by newser1** [comment]
I ran it both before and after turning the OSX firewall, and got the same error both times (curl: (60) SSL certificate problem: Invalid certificate chain).
bash-3.2$ /usr/local/opt/curl/bin/curl https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz -O
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (60) SSL certificate problem: Invalid certificate chain
More details here: http://curl.haxx.se/docs/sslcerts.html

curl performs SSL certificate verification by default, using a "bundle"
 of Certificate Authority (CA) public keys (CA certs). If the default
 bundle file isn't adequate, you can specify an alternate file
 using the --cacert option.
If this HTTPS server uses a certificate signed by a CA represented in
 the bundle, the certificate verification probably failed due to a
 problem with the certificate (it might be expired, or the name might
 not match the domain name in the URL).
If you'd like to turn off curl's verification of the certificate, use
 the -k (or --insecure) option.
It doesn't seem to be the firewall, since I'm getting the same error before and after turning it off.What is my next step?



    On Sunday, December 6, 2015 3:39 PM, Dominique <notifications@github.com> wrote:
 

 Does it work from the command line (i.e., no Homebrew involved)?curl https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz -O
or/usr/local/opt/curl/bin/curl https://downloads.sourceforge.net/project/lcms/lcms/2.7/lcms2-2.7.tar.gz -O
If not, please check your firewall settings.—

**Comment by tdsmith** [comment]
Homebrew uses the version of curl which is shipped by Apple, not Homebrew's. If you have a ~/.curlrc, try renaming it so that curl can't find it and see if the problem persists. If you don't have a .curlrc, it's likely you have a problem with your system certificate store, which I don't know how to fix.

**Comment by sysfu** [comment]
How can we force Homebrew to use the curl built from homebrew? I'm running into more and more instances where Apple's version of curl included in 10.6.8 will not download from web sites using TLS v1.1 or greater. 

For example when trying to install utox, I get the following error message:

```
curl: (35) Unknown SSL protocol error in connection to zodiaclabs.org:443 
```
If I use homebrew's curl (v7.47.1) to download the same file it works fine. 

Ssllabs shows that zodiaclabs.org does not support TLS v1.0 or lower, so it seems that Apple's curl v7.19.7 does not support TLS v1.1 or higher.

**Comment by dpo** [comment]
@sysfu that's a question for Homebrew core, but note that we no longer support 10.6.  I'll close this as it doesn't seem that anything can be done from here.
